Add allowAnyValue (#571)
JorelAli/CommandAPI: docssrc/src/argument_list.md

$$\downarrow$$

> ### Allowing any value (Optional)
>
> If you want to do your own validation you can enable the `allowAnyValue` method to set whether any value is allowed. By default, this is disabled.
>

> > When `allowAnyValue` is enabled, validation is skipped, and it is expected you will manually perform any validation required:

Reviewer: ```suggestion
> When `allowAnyValue` is set to `true`, list validation is skipped, and any value a player enters will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too:
```
Maybe this is a bit better?

Reviewer 2: I don't particularly like the phrase "list validation", we could just be clear as day and remove it. By saying "... is set to `true`, any value a player enters ...", it's so much cleaer.

> > When `allowAnyValue` is set to `true`, any value a player enter will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too

Reviewer: ```suggestion
> When `allowAnyValue` is set to `true`, any value a player enters will be allowed in the list regardless of the list provided for the argument. Note that duplicate values are still only permitted when `allowDuplicates` is set to `true` too
```

Reviewer 2: Full stop?

Reviewer: No, needs a colon here, I forgot that

>
> ```java
> ListArgumentBuilder.allowAnyValue(true);
> ```
>

> > ![List arguments with anyvalue enabled](./images/arguments/listargument_allow_any_values.gif)

Reviewer: ```suggestion
> ![List arguments with allowAnyValue set to true](./images/arguments/listargument_allow_any_values.gif)
```
